python/tests/test_variable_py_object.py

```diff
@@ -35,6 +35,8 @@ def test_scalar_Variable_py_object_change():
 
 def test_scalar_Variable_py_object_copy_is_deep_copy():
     var = sc.Variable(value=[1, 2, 3])
+    # Dataset.copy() releases the GIL. This will segfault unless
+    # scipp::python::PyObject::PyObject acquires the GIL.
```

Comment:
::~PyObject?

Reply:
No, I mean the copy constructor here. Destructor is relevant below in `__delitem__`.